hbase-server/src/main/java/org/apache/hadoop/hbase/util/HBaseFsck.java

```diff
@@ -800,7 +800,12 @@ public int onlineHbck()
       checkRegionBoundaries();
     }
 
-    checkAndFixReplication();
+    TableName hbaseReplicationTable =
+      TableName.valueOf(getConf().get(ReplicationStorageFactory.REPLICATION_QUEUE_TABLE_NAME,
+        ReplicationStorageFactory.REPLICATION_QUEUE_TABLE_NAME_DEFAULT.getNameAsString()));
+    if (admin.tableExists(hbaseReplicationTable)) {
```

Comment:
The ReplicationQueueStorage has a hasData method. I think we'd better make use of it.

I think we can introduce a should check method for ReplicationChecker class and call it in checkAndFixReplication.